daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1413,9 +1422,12 @@ private[lf] object SBuiltin {
         ),
         result = mbCoid,
         version = machine.tmplId2TxVersion(templateId),
-      )
-      checkAborted(onLedger.ptx)
-      machine.returnValue = SV.Unit
+      ) match {
+        case Left(err) => throw convTxError(err)
```

Comment:
We could factorize the `throw` inside `convTxError`. What do you think ?

Reply:
We could also have something like. 

```
def assertRight(result: Either[Tx.TransactionError, PartialTransaction]): PartialTransaction = result match {
   case Right(ptx) => ptx
   case Left(error) => throw convTxError(err)
}
```

Just suggestion, I do not feel strongly about that.

Reply:
I had a look. But I'm not sure it make the code any clearer. Especially as only 3 of the 4 cases can be factorized.
So I decided to leave it.

Actually, I wonder if a better direction would be to push the `throw` and `convTxError` down into the Ptx code.

(But again, that would require we avoid the annoying case of updating the `machine.ptx` in the error case.)